pkg/util/mocks/mock_types.go

```diff
@@ -0,0 +1,138 @@
+package mocks
+
+import (
+	"fmt"
+
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/kube/secret"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/util/authtypes"
+	"github.com/mongodb/mongodb-kubernetes-operator/pkg/util/errors"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	types "k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type MockSecretGetUpdateCreateDeleter struct {
+	secrets map[client.ObjectKey]corev1.Secret
+}
+
+func NewMockedSecretGetUpdateCreateDeleter(secrets ...corev1.Secret) secret.GetUpdateCreateDeleter {
+	mockSecretGetUpdateCreateDeleter := MockSecretGetUpdateCreateDeleter{}
+	mockSecretGetUpdateCreateDeleter.secrets = make(map[client.ObjectKey]corev1.Secret)
+	for _, s := range secrets {
+		mockSecretGetUpdateCreateDeleter.secrets[types.NamespacedName{Name: s.Name, Namespace: s.Namespace}] = s
+	}
+	return mockSecretGetUpdateCreateDeleter
+}
+
+func (c MockSecretGetUpdateCreateDeleter) DeleteSecret(objectKey client.ObjectKey) error {
+	delete(c.secrets, objectKey)
+	return nil
+}
+
+func (c MockSecretGetUpdateCreateDeleter) UpdateSecret(s corev1.Secret) error {
+	c.secrets[types.NamespacedName{Name: s.Name, Namespace: s.Namespace}] = s
+	return nil
+}
+
+func (c MockSecretGetUpdateCreateDeleter) CreateSecret(secret corev1.Secret) error {
+	return c.UpdateSecret(secret)
+}
+
+func (c MockSecretGetUpdateCreateDeleter) GetSecret(objectKey client.ObjectKey) (corev1.Secret, error) {
+	if s, ok := c.secrets[objectKey]; !ok {
+		return corev1.Secret{}, errors.NotFoundError()
+	} else {
+		return s, nil
+	}
+}
+
+type MockConfigurable struct {
+	opts   authtypes.Options
+	users  []authtypes.User
+	nsName types.NamespacedName
+	refs   []metav1.OwnerReference
+}
+
+func NewMockConfigurable(opts authtypes.Options, users []authtypes.User, nsName types.NamespacedName, refs []metav1.OwnerReference) *MockConfigurable {
```

Comment:
whats the reason to return a pointer here?

Reply:
I think I mistakenly autogenerated it with pointers. Will change it.